### ISSUE TITLE
Insert Ignore LastInsertedId is -1 when Ignored

### ISSUE THREAD

**Issue by mkehoe** [mysql-net/MySqlConnector#1147]
**Software versions**
MySqlConnector version: 2.1.7
Server type (MySQL, MariaDB, Aurora, etc.) and version: MySql 5.7
.NET version: 6.0

**Describe the bug**
When using an insert ignore statement with ExecuteNonQueryAsync(), LastInsertedId is -1 (expected 0) if the insert is ignored. I believe this is changed behavior from an earlier version.

**Code sample**
```var connection = new MySqlConnection(connectStr);
await connection.OpenAsync();

var sql = @"insert ignore into test.test_table (id, foo) values (@id, @foo)";
await using var cmd = new MySqlCommand(sql, connection);

cmd.Parameters.AddWithValue("@id", 2);
cmd.Parameters.AddWithValue("@foo", "Test");

cmd.CommandType = System.Data.CommandType.Text;
await cmd.ExecuteNonQueryAsync();
```

**Expected behavior**
MySqlConnector docs for LastInsertedId reference MySql [LAST_INSERTED_ID()](https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_last-insert-id) and they state:

 > If you use INSERT IGNORE and the row is ignored, the LAST_INSERT_ID() remains unchanged from the current value (or 0 is returned if the connection has not yet performed a successful INSERT)

This statement leads me to believe that 0 for LastInsertedId is correct in this scenario.

**Comment by bgrainger** [comment]
> I believe this is changed behavior from an earlier version.

It looks like the value of `cmd.LastInsertedId` (in this example) changed from 0 to -1 between 1.3.14 and 2.0.0.

**Comment by bgrainger** [comment]
Fixed in [2.1.8](https://github.com/mysql-net/MySqlConnector/releases/tag/2.1.8).
